refactor(transformer/class-properties): simplify to get `constructor` method only once

Dunqing opened the pull request:
Benefit from #10495, we can move getting `instance_inits_scope_id` and `instance_inits_constructor_scope_id` part to after all instance properties transformed. So that we can avoid to get `constructor` twice due to borrow checker.

Dunqing commented:
> [!WARNING]
> <b>This pull request is not mergeable via GitHub because a downstack PR is open. Once all requirements are satisfied, merge this PR as a stack <a href="https://app.graphite.dev/github/pr/oxc-project/oxc/10493?utm_source=stack-comment-downstack-mergeability-warning" >on Graphite</a>.</b>
> <a href="https://graphite.dev/docs/merge-pull-requests">Learn more</a>

* **#10493** <a href="https://app.graphite.dev/github/pr/oxc-project/oxc/10493?utm_source=stack-comment-icon" target="_blank"><img src="https://static.graphite.dev/graphite-32x32-black.png" alt="Graphite" width="10px" height="10px"/></a> 👈 <a href="https://app.graphite.dev/github/pr/oxc-project/oxc/10493?utm_source=stack-comment-view-in-graphite" target="_blank">(View in Graphite)</a>
* **#10492** <a href="https://app.graphite.dev/github/pr/oxc-project/oxc/10492?utm_source=stack-comment-icon" target="_blank"><img src="https://static.graphite.dev/graphite-32x32-black.png" alt="Graphite" width="10px" height="10px"/></a>
* `main`
<hr />

## How to use the Graphite Merge Queue

Add either label to this PR to merge it via the merge queue:
- <kbd>0-merge</kbd> - adds this PR to the back of the merge queue
- <kbd>hotfix</kbd> - for urgent hot fixes, skip the queue and merge this PR next

You must have a Graphite account in order to use the merge queue. Sign up using [this link](https://app.graphite.dev/invite/github/oxc-project?ref=merge-queue-instructions-in-stack-comment&prId=10493).

<h2></h2>
<sub>

An organization admin has enabled the [Graphite Merge Queue](https://graphite.dev/docs/graphite-merge-queue) in this repository.

</sub>

<sub>

Please do not merge from GitHub as this will restart CI on PRs being processed by the merge queue.

</sub>

<h2></h2>

This stack of pull requests is managed by <a href="https://graphite.dev?utm-source=stack-comment"><b>Graphite</b></a>. Learn more about <a href="https://stacking.dev/?utm_source=stack-comment">stacking</a>.
<!-- Current dependencies on/for this PR: -->

overlookmotel reviewed (approved):
Nice! Yes, getting constructor twice was necessary before only to work around the borrow checker. You've solved that problem now.

Dunqing committed:
refactor(transformer/class-properties): simplify to get `constructor` method only once (#10493)

Benefit from #10495, we can move getting `instance_inits_scope_id` and `instance_inits_constructor_scope_id` part to after all instance properties transformed. So that we can avoid to get `constructor` twice due to borrow checker.
